frontend/src/pages/Chats.tsx

```diff
@@ -1,9 +1,83 @@
 import TopAppBar from '../components/TopAppBar';
+import { useEffect, useState } from 'react';
+
+import { StreamChat } from 'stream-chat';
+import {
+  Chat,
+  Channel,
+  Window,
+  ChannelHeader,
+  MessageList,
+  MessageInput,
+  Thread,
+  LoadingIndicator,
+  ChannelList,
+} from 'stream-chat-react';
+
+import 'stream-chat-react/dist/css/index.css';
+import './Chats.css';
+
+const apiKey = process.env.REACT_APP_STREAM_API_KEY
+  ? process.env.REACT_APP_STREAM_API_KEY
+  : '';
+
+const user = {
+  id: 'john',
+  name: 'John',
+  image: 'https://getstream.imgix.net/images/random_svg/FS.png',
+};
+
+const filters = { type: 'messaging', members: { $in: [user.id] } };
+const sort = [{ last_message_at: -1 as const }];
 
 export const Chats = () => {
+  const [client, setClient] = useState<any>(null);
+  const [channel, setChannel] = useState<any>(null);
+
+  useEffect(() => {
+    async function init() {
+      const chatClient = StreamChat.getInstance(apiKey);
+
+      // Dev Token for now, would need to use appSecret to create real userToken
+      await chatClient.connectUser(user, chatClient.devToken(user.id));
+
+      const channel = chatClient.channel('messaging', 'react-talk', {
+        image: 'https://www.drupal.org/files/project-images/react.png',
+        name: 'Talk about React',
+        members: [user.id],
+      });
+
+      await channel.watch();
+
+      setChannel(channel);
+      setClient(chatClient);
```

Comment:
We should try to keep tsx files only for visible components as much as possible (i.e. buttons, textfields, message lists etc.)

In `frontend/src/functions` let's create a file called `chats.ts`. Here we can get the channel and the client. I'd also like for that function to return a type rather than an any - that way line 34 and 35 can set a type. (A tip to find out what type it is - go to VSCode, hover over the constant name e.g. the `chatClient` in line 39 and VSCode should show you something like this `const chatClient: StreamChat<DefaultGenerics>`).

In [this](https://linear.app/uni-connector/issue/UNI-36/investigate-how-to-connect-the-chat-client-to-our-backend) card, where we connect the backend to the chat client, the goal is that we _only_ change the chats file rather than any tsx file

Reply:
I've abstracted away the getting the client and the channel. 

For types, firstly while the `Chat` type is working, the `Channel` type is not obvious. If you hover over it, it gives you `Channel<DefaultGenerics>` but that doesn't work for some reason, it raises an error. 

Also changing `<any>` to `StreamChat<DefaultGenerics>` for the client also raises an error with the `useEffect` part which I'm not sure about.

Finally, there is still some logic in the frontend - namely the `user` part - I'm assuming that when we connect the backend, the frontend will send the userID it gets when the user authenticates to the backend which will deal with things there?

Reply:
Adding a return type to the `getChannel` function of `Channel<DefaultGenerics>` seems to be working for me. Are you sure you're importing the `Channel` type from `stream-chat`?

Reply:
If you change the any to the stream chat type, it'll error because you're setting the default value to `null`. `null` is not compatible with `StreamChat<DefaultGenerics>`. React actually has a neat solution for this, you can just do something like this:
```
const [client, setClient] = useState<StreamChat<DefaultGenerics>>();
```

Now your client will be either `StreamChat<DefaultGenerics>` OR it will be the default value of `undefined` (which is what `()` signifies).

Javascript has a difference between `null` and `undefined` which can be a little confusing BUT in both cases you can check if the value exists simply by doing this:
```
if (client) {
  // client exists
}
else {
  // client is null or undefined
}
```

Reply:
> Finally, there is still some logic in the frontend - namely the user part - I'm assuming that when we connect the backend, the frontend will send the userID it gets when the user authenticates to the backend which will deal with things there?

That's correct. When you hit the 'sign in' button, you authenticate with Microsoft who generate a token for us. Our backend will decode that token and within that token, will be an ID. If you are interested, [this](https://github.com/Siyer2/uni-connector/blob/0e1feb4ae5ff42206da014c088bbe2a770ee8cd3/backend/src/middleware/index.ts#L44) is where the backend decodes the token. And [this](https://github.com/Siyer2/uni-connector/blob/0e1feb4ae5ff42206da014c088bbe2a770ee8cd3/backend/src/middleware/types.ts#L14) is the type that is returned from the decoded token (`oid` is the userId).

Also, if you'd like to see how our frontend is calling our backend, we are using a library called Axios. [Here](https://github.com/Siyer2/uni-connector/blob/1786360c6558bf76f4420da6b48c2a8d419b4563/frontend/src/api/index.ts#L4) is how we're doing it.